hermes-management/src/test/groovy/pl/allegro/tech/hermes/management/infrastructure/metrics/HybridPrometheusBasedSubscriptionMetricsRepositoryTest.groovy

```diff
@@ -22,29 +22,34 @@ class HybridPrometheusBasedSubscriptionMetricsRepositoryTest extends Specificati
-            "|hermes_consumers_subscription_timeouts_total" +
-            "|hermes_consumers_subscription_retries_total" +
-            "|hermes_consumers_subscription_throughput_bytes_total" +
-            "|hermes_consumers_subscription_other_errors_total" +
-            "|hermes_consumers_subscription_batches_total" +
-            "|hermes_consumers_subscription_http_status_codes_total', " +
-            "group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]) keep_metric_names)"
+    private final static MetricsQuery deliveredQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_delivered_total', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))")
+    private final static MetricsQuery timeoutsQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_timeouts_total', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))")
+    private final static MetricsQuery retriesQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_retries_total', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))")
+    private final static MetricsQuery throughputQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_throughput_bytes_total', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))")
+    private final static MetricsQuery otherErrorsQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_other_errors_total', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))")
+    private final static MetricsQuery batchesQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_batches_total', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))")
+    private final static MetricsQuery status2xxQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_http_status_codes_total', group='group', topic='topic', subscription='subscription', status_code=~'2.*', service=~'hermes'}[1m]))")
+    private final static MetricsQuery status4xxQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_http_status_codes_total', group='group', topic='topic', subscription='subscription', status_code=~'4.*', service=~'hermes'}[1m]))")
+    private final static MetricsQuery status5xxQuery = new MetricsQuery("sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_http_status_codes_total', group='group', topic='topic', subscription='subscription', status_code=~'5.*', service=~'hermes'}[1m]))")
```

Comment:
As far as I can see all queries differs only by a little. Maybe we would be able to create something more readable like:
```
private final static String subscriptionQuery = "sum by (group, topic, subscription) (irate({__name__='hermes_consumers_subscription_%s', group='group', topic='topic', subscription='subscription', service=~'hermes'}[1m]))";

private final static MetricsQuery timeoutsQuery = new MetricsQuery(String.format(subscriptionQuery, "timeouts"));
private final static MetricsQuery deliveredQuery = new MetricsQuery(String.format(subscriptionQuery, "delivered"));
...
```
What do You think about it?

Reply:
I wanted to somewhat make the query independent of a specific provider, but now I don't see any benefits so I can delete it.

Reply:
fixed